### PR TITLE
1011/feat/publisher

### PR DESCRIPTION
**Pull request by ac0stas**
## Proposed changes

Added functionality to create publisher nodes in the blockchain network

**Trello Ticket** [here](https://trello.com/c/mMz2aPH5/21-10101-feature-create-action-publisher)

## Types of changes

What types of changes does your code introduce to this Blockchain?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added necessary documentation (if appropriate)
- [x] My pull request name and branch name it's with the Trello Ticket format

### CODE REVIEW

File: commandLine/options.go
Comment (benja-iag) [inline commandLine/options.go:1]: Please delete the commented lines

File: commandLine/options.go [hunk 1]
```diff
@@ -159,3 +162,41 @@ func createWallet(cmd *cobra.Command, args []string) {
...
+func createPublisher(cmd *cobra.Command, args []string) {
+	for {
+		nodeInfo := utils.GetNodeInfo()
+			if nodeInfo == nil {
+				fmt.Println("'port.pid' file already exists, cannot create a publisher")
+			} else {
+				fmt.Println("port.pid file exists, proceeding to start publisher node.")
+				network.P2p(true) 
+		}
+	}
```
Comment (benja-iag) [inline commandLine/options.go]: Please delete the for, this part of the code must be do it just one time
```suggestion
		nodeInfo := utils.GetNodeInfo()
			if nodeInfo == nil {
				fmt.Println("'port.pid' file already exists, cannot create a publisher")
			} else {
				fmt.Println("port.pid file exists, proceeding to start publisher node.")
				network.P2p(true) 
		}
```

File: commandLine/options.go [hunk 2]
```diff
@@ -159,3 +162,41 @@ func createWallet(cmd *cobra.Command, args []string) {
...
+	
+}
+
+func createSubscriber(cmd *cobra.Command, args []string) {
+	fmt.Println("Starting subscriber node...")
+	for {
+		
+		network.P2p(false)	
+	}
```
Comment (benja-iag) [inline commandLine/options.go]: Please delete the for, this part of the code must be do it just one time
```suggestion
		
		network.P2p(false)	
```

**Review by benja-iag — CHANGES_REQUESTED** [review]

File: commandLine/options.go
Comment (ac0stas) [reply commandLine/options.go:1]: The comment has already been corrected

File: commandLine/options.go (hunk 1 above)
Comment (ac0stas) [reply commandLine/options.go]: The comment has already been corrected

File: commandLine/options.go (hunk 2 above)
Comment (ac0stas) [reply commandLine/options.go]: The comment has already been corrected

File: commandLine/options.go [hunk 3]
```diff
@@ -159,3 +162,19 @@ func createWallet(cmd *cobra.Command, args []string) {
...
+func createPublisher(cmd *cobra.Command, args []string) {
+	nodeInfo := utils.GetNodeInfo()
+	if nodeInfo == nil {
+		fmt.Println("'port.pid' file already exists, cannot create a publisher")
+
+	} else {
+		fmt.Println("port.pid file exists, proceeding to start publisher node.")
+		network.P2p(true) 
+	}
+}
```
Comment (benja-iag) [inline commandLine/options.go]: The if - else statement should be:
If the file port.pid already exist, the function shouldn't call the network.P2p function.
If the file port.pid doesn't exist, the function should create a publisher node.

**Review by benja-iag — CHANGES_REQUESTED** [review]

**Pull request merged**